include/zephyr/posix/sys/stat.h

```diff
@@ -0,0 +1,270 @@
+/*
+Copyright (c) 1982, 1986, 1993
+The Regents of the University of California.  All rights reserved.
```

Comment:
3-Clause BSD license should be acceptable here, but it will require TSC approval.